framework/auth/views.py

```diff
@@ -184,9 +184,10 @@ def confirm_email_get(token, auth=None, **kwargs):
                 return redirect(
                     campaigns.campaign_url_for(campaign)
                 )
-            if len(auth.user.emails) == 1:
+            if len(auth.user.emails) == 1 and len(auth.user.email_verifications) == 0:
```

Comment:
Discussed in person: this is actually correct. At this point, the user's primary email will have been confirmed, so there should be no email_verifications.